### PR TITLE
Utility class with methods to handle HTTP headers

### PR DESCRIPTION
**Pull request by ayurchuk**
Removed JavaStreams class from `ts-reactive-marshal` in favor of duplicate in `ts-reactive-java`.
Changed order of parameters in WriteProtocol to be have the same order as in other write protocols.

@jypma, @alar17 please review.

### CODE REVIEW

File: ts-reaktive-marshal-akka/src/main/java/com/tradeshift/reaktive/marshal/http/Directives.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+import java.net.URI;
+import java.util.function.Supplier;
+
+import akka.http.javadsl.model.headers.Location;
+import akka.http.javadsl.server.Route;
+
+/**
+ * Various functions relating to Akka HTTP headers
+ */
+public class Directives {
```
Comment (jypma) [inline ts-reaktive-marshal-akka/src/main/java/com/tradeshift/reaktive/marshal/http/Directives.java]: this class is unrelated to marshalling, so should be in the `ts-reaktive-akka` module.

File: ts-reaktive-marshal-akka/src/test/java/com/tradeshift/reaktive/marshal/http/RouteTest.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+import akka.actor.ActorSystem;
+import akka.http.javadsl.model.HttpRequest;
+import akka.http.javadsl.server.RouteResult;
+import akka.http.javadsl.testkit.TestRouteResult;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+
+@RunWith(CuppaRunner.class)
+public class RouteTest extends akka.http.javadsl.testkit.RouteTest {
```
Comment (jypma) [inline ts-reaktive-marshal-akka/src/test/java/com/tradeshift/reaktive/marshal/http/RouteTest.java]: let's not drop this class in here until we've verified that this hack is still necessary. Try running some of our route tests with the plain akka `JUnitRouteTest`. It used to have a magic dependency on `scalatest`, perhaps that's been resolved since.

Comment (jypma) [inline ts-reaktive-marshal-akka/src/test/java/com/tradeshift/reaktive/marshal/http/RouteTest.java]: The class is also not generic enough for ts-reaktive, since it has a hardcoded config.

File: ts-reaktive-marshal/src/main/java/com/tradeshift/reaktive/json/JSONProtocol.java [hunk 3]
```diff
@@ -149,9 +151,55 @@
     /**
      * Returns a write-only protocol for a JSON object with fields [p*], using [g*] to get the fields when writing.
      */
-    public static <F1,F2,F3,T> ObjectWriteProtocol<T> object(WriteProtocol<JSONEvent, F1> p1, Function1<T, F1> g1, WriteProtocol<JSONEvent, F2> p2, Function1<T, F2> g2, WriteProtocol<JSONEvent, F3> p3, Function1<T, F3> g3) {
+    public static <F1,F2,F3,T> ObjectWriteProtocol<T> object(Function1<T, F1> g1, WriteProtocol<JSONEvent, F1> p1, Function1<T, F2> g2, WriteProtocol<JSONEvent, F2> p2, Function1<T, F3> g3, WriteProtocol<JSONEvent, F3> p3) {
```
Comment (jypma) [inline ts-reaktive-marshal/src/main/java/com/tradeshift/reaktive/json/JSONProtocol.java]: hmm were these reversed? how strange. We should really codegen this. If you get bored next week, [this](https://github.com/sbt/sbt-boilerplate) is what the akka guys are using, feel free to give it a shot.

Comment (ayurchuk) [inline ts-reaktive-marshal/src/main/java/com/tradeshift/reaktive/json/JSONProtocol.java]: Cool! I will check this plugin.

File: ts-reaktive-marshal-akka/src/main/java/com/tradeshift/reaktive/marshal/http/Directives.java (hunk 1 above)
Comment (ayurchuk) [inline ts-reaktive-marshal-akka/src/main/java/com/tradeshift/reaktive/marshal/http/Directives.java]: I will move it.

File: ts-reaktive-marshal-akka/src/test/java/com/tradeshift/reaktive/marshal/http/RouteTest.java (hunk 2 above)
Comment (ayurchuk) [inline ts-reaktive-marshal-akka/src/test/java/com/tradeshift/reaktive/marshal/http/RouteTest.java]: I added `JUnitRouteTest` with `scalatest` dependency, because ticket is open at the moment: https://github.com/akka/akka-http/issues/216.

File: ts-reaktive-akka/src/main/java/com/tradeshift/reaktive/akka/Directives.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    /**
+     * Adds a Location header containing the absolute path, by resolving [relativePath] against the "matched path"
+     * of the current request as base URI. The base URI is always extended to end in "/"; in other words, the matched
+     * path is assumed to indicate a directory under which [relativePath] is to be resolved.
+     *
+     * @param relativePath The relative path for the location header. Will be resolved relative to the currently matched path.
+     */
+    public static Route addLocationHeader(String relativePath, Supplier<Route> inner) {
+        return extractRequestContext(ctx -> {
```
Comment (jypma) [inline ts-reaktive-akka/src/main/java/com/tradeshift/reaktive/akka/Directives.java]: could you try if this would be simpler using [extractMatchedPath](http://doc.akka.io/docs/akka-http/10.0.4/java/http/routing-dsl/directives/basic-directives/extractMatchedPath.html) ? That should directly give us `base`.

**Comment by jypma** [comment]
@ayurchuk  Shall I do a release with this in it?
